org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/analysis/constraint/AbstractConstraintAnalyzer.java

```diff
@@ -157,6 +205,32 @@ private boolean isEmptyAST(IStrategoTerm ast) {
         return Tools.isTermTuple(ast) && ast.getSubtermCount() == 0;
     }
 
+    protected IStrategoTerm getFlowSpecTransferFunctions(ILanguageComponent component) {
+        IStrategoTerm transferFunctions = flowSpecTransferFunctionCache.get(component);
+        if (transferFunctions != null) {
+            return transferFunctions;
+        }
+
+        FileObject tfs = resourceService.resolve(component.location(), TRANSFER_FUNCTIONS_FILE);
```

Comment:
How does this behave for languages without this file?

Reply:
It throws an Exception. I can turn it into an optional instead if you prefer.

Reply:
What I mean is, does this crash on languages that use NaBL2 but not Flowspec? I don't know when/where this function is called. I guess a graceful failure is usually preferable over runtime exceptions.

Reply:
This method gets called when the NaBL2 solver produces a non-empty CFG, meaning that FlowSpec was used in the project. FlowSpec should then have also generated the transfer functions file, so ideally the runtime exception just doesn't occur. But I'll just leave the logger statement, use optionals, and ignore the case where no file is present.